packages/slice-machine/src/features/onboarding/types.ts

```diff
@@ -17,9 +21,17 @@ export type OnboardingStepContentDefinition = {
   videoUrl: string;
 };
 
+export const onboardingStepStatusesSchema = z.object(
+  Object.fromEntries(
+    onboardingSteps.map((step) => [step.id, z.boolean()]),
+  ) as Record<OnboardingStepType, z.ZodBoolean>,
```

Comment:
No need to cast here. Create an empty z.object({}), then iterate over each step and extend the initial schema with the id and the boolean. Something like:
```
schema = schema.extend({
              [key]: z.boolean(),
            })
```

Reply:
Solved after a quick chat to understand the suggestion better ✅